### ISSUE TITLE
Heapster can't find Node IP Addresses

### ISSUE THREAD

**Issue by yossaa** [Azure/AKS#665]
**What happened**:
Heapster reports error every minute:
E0926 11:25:05.000480       1 summary.go:404] node aks-nodepool1-10072107-1 has no valid hostname and/or IP address: aks-nodepool1-10072107-1
E0926 11:26:05.000464       1 summary.go:404] node aks-nodepool1-10072107-1 has no valid hostname and/or IP address: aks-nodepool1-10072107-1
E0926 11:27:05.042431       1 summary.go:404] node aks-nodepool1-10072107-1 has no valid hostname and/or IP address: aks-nodepool1-10072107-1

Heapster tries to get IP from node description, but it contains only Hostname, no IP:
kubectl describe node aks-nodepool1-10072107-1
Name:               aks-nodepool1-10072107-1
Roles:              agent
...
Addresses:
  **Hostname:  aks-nodepool1-10072107-1**
Capacity:

**What you expected to happen**:
No idea, but it's weird that metrics service generates such errors all the time

**How to reproduce it (as minimally and precisely as possible)**:
check heapster logs:
kubectl logs -f heapster-6c9777d684-xxxx heapster -n kube-system

**Environment**:
- Kubernetes version (use `kubectl version`): 1.11.2
- Size of cluster (how many worker nodes are in the cluster?) 1

**Comment by yossaa** [comment]
Same error is reported by "metrics-server" pods

**Comment by choovick** [comment]
Just got the same issue. Possibly related to https://github.com/Azure/acs-engine/issues/3503.

Confirm issue that affected node does not have IP:

`kubectl describe node <node_name> | grep 'Addresses:' -A 2`

Fix affected node:

```
kubectl drain <node_name> --ignore-daemonsets

# reboot VM in dashboard or via ssh

kubectl uncordon <node_name>
```

Delete metric server pod and schedule will reschedule it... Error is gone for now.

**Comment by feiskyer** [comment]
The fix (https://github.com/kubernetes/kubernetes/pull/70400) should be landed in kubernetes 1.11.6
